pkg/rules/rules.go

```diff
@@ -80,8 +82,57 @@ func (rr *GRPCClient) Rules(ctx context.Context, req *rulespb.RulesRequest) (*ru
 	return &rulespb.RuleGroups{Groups: resp.groups}, resp.warnings, nil
 }
 
-// filterRules filters rules in a group according to given matcherSets.
-func filterRules(ruleGroups []*rulespb.RuleGroup, matcherSets [][]*labels.Matcher) []*rulespb.RuleGroup {
+// filters rules by group name, rule name or file.
+func filterRulesByNamesAndFile(ruleGroups []*rulespb.RuleGroup, ruleName []string, ruleGroup []string, file []string) []*rulespb.RuleGroup {
+	if len(ruleName) == 0 && len(ruleGroup) == 0 && len(file) == 0 {
+		return ruleGroups
+	}
+
+	queryFormToSet := func(values []string) map[string]struct{} {
+		set := make(map[string]struct{}, len(values))
+		for _, v := range values {
+			set[v] = struct{}{}
+		}
+		return set
+	}
+
+	rnSet := queryFormToSet(ruleName)
+	rgSet := queryFormToSet(ruleGroup)
+	fSet := queryFormToSet(file)
```

Comment:
Do you think just using `slices.Contains` instead of set might be more memory intensive?

Reply:
The reason I did it like this, was simply because that's how the code looks like in Prometheus. I am not sure if there are significant performance benefits either way, although perhaps the code is a bit cleaner using `slices.Contains`, so happy to change to that approach if you think it's best?

Reply:
Ah I see, fine if it mirrors prometheus then